### DISCUSSION TITLE
Security considerations

### DISCUSSION THREAD

**Discussion by jakobvase** [pkgxdev/pkgx#354] [Ideas]
Hi! Awesome project, I really want to use it!

I have one worry that keeps me from using the magic (and fun!) part of tea. If any command I use that doesn't exist on my machine causes a random package to be run, isn't that a huge security risk? What if I misspell `git`, and accidentally run `got push` and `got` is a malicious package that pushes my code to a place I can't control?

Thanks for all your hard work!

**Comment by getneil** [comment]
maybe a scenario like this  would work to prompt the user, here is the scenario im imagining

HAPPY PATH
```bash
$ git add ./file.x
tea: would you like to use `git-scm.org`? y/n. Y
tea: would remember this decision till eternity. to edit do `tea -undo git`
```
then tea might remember this decision in the future

NOT SO HAPPY PATH
```bash
$ gitt add ./file.x
tea: would you like to use `shady-gitt.com`? y/n. N

**Reply by jakobvase** [reply]
Yeah and possibly with a whitelist of some sort with very common packages. Would be great!

**Comment by mxcl** [comment]
Yes, I have thought about this.

So partly: you trust us not to package typo squatters that can be damaging. tea will only install things we vetted. We check submissions for legitimacy.

And yes, we're considering having you agree before we install something via magic. The prompt could offer to not ask again so you can gauge your trust level.

And of course: magic is opt in and optional.

**Reply by jakobvase** [reply]
Thanks for replying! I'm happy to hear that you're already vetting. That's my concern gone.
